WhatToCook.Application.Tests/Services/MealPlanningServiceTests.cs

```diff
@@ -0,0 +1,197 @@
+                new() { Name = "OldRecipe1" },
+                new() { Name = "OldRecipe2" }
+            }
+        };
+
+        //setup GetMealPlanByName method to return existing plan of meals
+        _mealPlanningRepositoryMock.Setup(x => x.GetMealPlanByName(It.IsAny<string>())).ReturnsAsync(existingPlanOfMeals);
+
+        var recipes = new List<Recipe>()
+        {
+            new() { Name = "Recipe1" },
+
+            new() { Name = "Recipe2" }
+        };
+        //setup GetByNames method to return new recipes
+        _recipesRepositoryMock.Setup(x => x.GetByNames(It.IsAny<IEnumerable<string>>())).Returns(recipes);
+        //arrange instances of interfaces
+        var sut = new MealPlanningService(_recipesRepositoryMock.Object, _mealPlanningRepositoryMock.Object);
+
+        //arrange new plan of meal data
+        var updatePlanOfMealRequest = new UpdatePlanOfMealRequest()
+        {
+            Name = "NewTest",
+            FromDate = DateTime.UtcNow,
+            ToDate = DateTime.UtcNow.AddDays(2),
+            Recipes = recipes
+        };
+        // _mealPlanningRepositoryMock.Setup(x => x.Update(It.IsAny<PlanOfMeals>())).Verifiable();
+
+        //Act
+        //update plan of meal with new data
+        var result = await sut.Update(updatePlanOfMealRequest);
+
+        //Assert
+        result.Should().NotBeNull();
+        result.Name.Should().NotBeNullOrWhiteSpace().And.Be(updatePlanOfMealRequest.Name);
+        result.Recipes.Should().NotBeNullOrEmpty().And.BeEquivalentTo(recipes);
+
+        _mealPlanningRepositoryMock.Verify();
+    }
+    [Fact]
+    public async Task Given_GreaterToDateThanFromDate_When_UpdatingMealPlan_ThenShouldThrowException()
+    {
+        // Arrange
+        var existingPlanOfMeals = new PlanOfMeals()
+        {
+            Id = 1,
+            Name = "OldTest",
+            FromDate = DateTime.UtcNow.AddDays(0),
+            ToDate = DateTime.UtcNow.AddDays(1),
+            Recipes = new List<Recipe>()
+        {
+            new() { Name = "OldRecipe1" },
+            new() { Name = "OldRecipe2" }
+        }
+        };
+
+        _mealPlanningRepositoryMock.Setup(x => x.GetMealPlanByName(It.IsAny<string>())).ReturnsAsync(existingPlanOfMeals);
+
+        var recipes = new List<Recipe>()
+        {
+            new() { Name = "Recipe1" },
+            new() { Name = "Recipe2" }
+        };
+
+        _recipesRepositoryMock.Setup(x => x.GetByNames(It.IsAny<IEnumerable<string>>())).Returns(recipes);
+
+        var sut = new MealPlanningService(_recipesRepositoryMock.Object, _mealPlanningRepositoryMock.Object);
+
+        var updatePlanOfMealRequest = new UpdatePlanOfMealRequest()
+        {
+            Name = "NewTest",
+            FromDate = DateTime.UtcNow.AddDays(7),
+            ToDate = DateTime.UtcNow.AddDays(-1),
+            Recipes = recipes
+        };
+
+        // Act & Assert
+        await Assert.ThrowsAsync<Exception>(() => sut.Update(updatePlanOfMealRequest));
```

Comment:
You could use fluent assertions like in other places. More info on how to do it can be found here:
https://fluentassertions.com/exceptions/

Reply:
done, will be in the next commit